esp/platform/espprotocol.cpp

```diff
@@ -68,6 +68,7 @@ CEspApplicationPort::CEspApplicationPort(bool viewcfg, CEspProtocol* prot) : bin
 
 void CEspApplicationPort::appendBinding(CEspBindingEntry* entry, bool isdefault)
 {
+    WriteLockBlock wblock(rwLock);
     bindings[bindingCount]=entry;
```

Comment:
There should be a check that bindingCount < max elements (currently a constant 100).  I don't know how likely we are to hit that limit....